### ISSUE TITLE
Small issues that hinder recording gameplay: video layer extends beyond actual image, impossible to set exact size in window mode, not all resolutions available in fullscreen. 

### ISSUE THREAD

**Issue by mzso** [stenzek/duckstation#2787]
**Game details**
Not game related

**Description of the issue / bug**
I like to record gameplay sometimes. There are some issues with this when using duckstation:
* I can't set an exact video size in windowed mode. I found no setting*, but there's also no feedback on the GUI either, to manually reach a particular size. There's also the pretty common ability to allow only sizes that match the video aspect ratio, which is also not used.
* Apart from, but related to this: there's this thing that the video is rendered letterboxed, either on top/bottom, or left right. Not sure if there's a technical limitation (I guess no because the video can be resized dynamically), but I think it would be best if the video surface should match the area where things are actually drawn, and the rest remaining some blank part of the UI. 
* For full screen I get this ultra long list. But still I can't find every resolution. I wanted to try 1920*1440. There's 2560*1440, but it results in a letterboxed video recording. In old epsxe/pcsx-r (Pete's opengl plugin) I can set a resolution of 2560*1440 with a "window size" of 1920*1440 (plus widescreen fix enabled), and I get proper 4:3 ratio video recorded with 1920*1440 resolution. Not sure how was that done there.

**Controller Troubleshoot Report**
Not relevant.

**Steps to reproduce**
[Try to provide as much detail as possible to reproduce the issue]

**Enhancements information**
[Make sure they are all turned off before report]

**Hardware/software**
Ryzen5 1600, RX580, Win10

**Emulator version**
0.1-4854-gbee50481 (master)

**Additional context**
[Add any other context about the problem here]

***Edit:**
So I found the "window size" option, in "view". Though it's not really deterministic, since it results in different window sizes when I use the setting in the game menu or during gameplay. It's more like a workaround that sometimes may serve.

**Comment by stenzek** [comment]
That's because games don't render at a common resolution, it varies, hence the integer scaled window size will also vary.

**Comment by mzso** [comment]
[**@stenzek**](https://github.com/stenzek) commented on [2021. dec. 31. 15:02 CET](https://github.com/stenzek/duckstation/issues/2787#issuecomment-1003383724 "2021-12-31T14:02:14Z - Replied by Github Reply Comments"):
> That's because games don't render at a common resolution, it varies, hence the integer scaled window size will also vary.

Indeed. But that's the least of the multiple aspects I brought up.
* _The window could keep to the rendered aspect ratio when resizing._ (Most media players work like this in windowed mode) An option might be added to set a static resolution for the window, beside x times options. DS can plainly do arbitrary size in real time
  * The GUI could at least have a feedback of the size of the video rectangle.
* In full screen borderless the graphical surface might also stick to the rendered display aspect ratio, so capture software wouldn't record a letterboxed video.
  * (I still don't get why old stuff like pcsx can do normal fullscreen 1920*1440, while DS can't.  Sure the resolution shows up 2560*1440, but the recording software records at 1920*1440 and shows osd accordingly as if the video is this resolution. Is there some padding trick that doesn't affect the graphical layer rectangle? It defies logic.)

I think these suggestions would be practical, doesn't seem like huge things to consider implementing.

PS: The _1-9x times resolution is rather unreliable_ even withing the same context, I get differing results even when I load the same save state, I got wacky resolutions such as 1908*1430 or 2276*1708 when I could get 1920*1440 before. It's weird. I had to reboot, at least the boot screen scaled predictably.
